### PR TITLE
Improve the concurrency support for EWF, and new standalone connector

### PR DESCRIPTION
**Pull request by jocelyn**
DO NOT MERGE THIS PULL REQUEST.
This is mainly use to show difference, and review the new design with WSF_EXECUTION.
Keep in mind this move is to add concurrency (and SCOOP) support.

See the MIGRATION.md note in new branch, to get an overview of changes.
There is still an effort to be done to improve comments, notes, and so on, especially for the examples.

Thanks.

### CODE REVIEW

File: library/server/ewsgi/connectors/cgi/src/wgi_cgi_connector.e [hunk 1]
```diff
@@ -67,6 +50,23 @@ feature -- Execution
 			end
 		end
 
+	process_rescue (res: detachable WGI_RESPONSE)
+		do
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/cgi/src/wgi_cgi_connector.e]: Missing description.

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/httpd_connection_handler.e [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+				process_handler (h)
+			else
+				check is_not_full: False end
+			end
+			debug ("dbglog")
+				dbglog (generator + ".LEAVE accept_connection_on_pool")
+			end
+		end
+
+	process_handler (hdl: separate HTTPD_REQUEST_HANDLER)
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/httpd_connection_handler.e]: Missing description.

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/httpd_request_handler.e [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+		end
+
+	CONCURRENT_POOL_ITEM
+		rename
+			release as release_pool_item
+		end
+
+feature {CONCURRENT_POOL, HTTPD_CONNECTION_HANDLER_I} -- Basic operation		
+
+	release
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/httpd_request_handler.e]: Missing description

File: library/server/ewsgi/connectors/libfcgi/src/wgi_libfcgi_connector.e [hunk 4]
```diff
@@ -1,24 +1,29 @@
...
-	make
+	SHARED_HTML_ENCODER
+		redefine
+			default_create
+		end
 
 feature {NONE} -- Initialization
 
-	make (a_service: like service)
+	default_create
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/libfcgi/src/wgi_libfcgi_connector.e]: Missing description.

File: library/server/ewsgi/connectors/libfcgi/src/wgi_libfcgi_connector.e [hunk 5]
```diff
@@ -88,6 +82,23 @@ feature -- Execution
 				retry
 			end
 		end
+	
+	process_rescue (res: detachable WGI_RESPONSE)
+		do
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/libfcgi/src/wgi_libfcgi_connector.e]: Missing Description.

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/none/httpd_connection_handler.e [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+feature {NONE} -- Initialization
+
+	initialize
+		do
+		end
+
+feature -- Access
+
+	is_shutdown_requested: BOOLEAN
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/none/httpd_connection_handler.e]: Missing Description.

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+	make (n: INTEGER)
+		do
+			capacity := n
+			create items.make_empty (n)
+			create busy_items.make_empty (n)
+		end
+
+feature -- Access
+
+	count: INTEGER
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e]: Missing Description.

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+		do
+			Result := count >= capacity
+		end
+
+	is_empty: BOOLEAN
+		do
+			Result := count = 0
+		end
+
+	capacity: INTEGER
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e]: Missing description.

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+	stop_requested: BOOLEAN
+
+feature -- Access
+
+	separate_item (a_factory: separate CONCURRENT_POOL_FACTORY [G]): detachable separate G
+		require
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e]: Item from a pool, if any.

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+					dbglog ("Released pool item #" + i.out + " (free:"+ (capacity - count).out +"))")
+				end
+			else
+				check known_item: False end
+			end
+		end
+
+feature -- Change
+
+	set_count (n: INTEGER)
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e]: is set_count or is set_capacity?

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+		local
+			g: detachable separate G
+		do
+			capacity := n
+			items.fill_with (g, 0, n - 1)
+			busy_items.fill_with (False, 0, n - 1)
+		end
+
+	terminate
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e]: Missing description.

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+		local
+			l_items: like items
+		do
+			l_items := items
+			l_items.wipe_out
+		end
+
+feature {NONE} -- Implementation
+
+	register_item (a_item: separate G)
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e]: Missing description.

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool_item.e [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+
+	set_pool (p: like pool)
+			-- Set associated `pool' to `p'.
+		do
+			pool := p
+		end
+
+feature {CONCURRENT_POOL, HTTPD_CONNECTION_HANDLER_I} -- Basic operation
+
+	release
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool_item.e]: Missing description.

File: library/server/ewsgi/connectors/standalone/src/implementation/wgi_httpd_request_handler_factory.e [hunk 14]
```diff
@@ -0,0 +1,41 @@
+note
+	description: "Summary description for {WGI_HTTPD_REQUEST_HANDLER_FACTORY}."
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/implementation/wgi_httpd_request_handler_factory.e]: Missing description.

File: library/server/ewsgi/connectors/standalone/src/implementation/wgi_httpd_request_handler_factory.e [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+
+class
+	WGI_HTTPD_REQUEST_HANDLER_FACTORY [G -> WGI_EXECUTION create make end]
+
+inherit
+	HTTPD_REQUEST_HANDLER_FACTORY
+
+feature -- Access
+
+	connector: detachable separate WGI_STANDALONE_CONNECTOR [G]
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/implementation/wgi_httpd_request_handler_factory.e]: Missing description.

File: library/server/ewsgi/connectors/standalone/src/implementation/wgi_httpd_request_handler_factory.e [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+feature -- Element change
+
+	set_connector (conn: like connector)
+		do
+			connector := conn
+		end
+
+feature -- Factory
+
+	new_handler: separate WGI_HTTPD_REQUEST_HANDLER [G]
```
Comment (jvelilla) [inline library/server/ewsgi/connectors/standalone/src/implementation/wgi_httpd_request_handler_factory.e]: Missing description

File: library/server/ewsgi/connectors/cgi/src/wgi_cgi_connector.e (hunk 1 above)
Comment (jocelyn) [inline library/server/ewsgi/connectors/cgi/src/wgi_cgi_connector.e]: done

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/none/httpd_connection_handler.e (hunk 6 above)
Comment (jocelyn) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/none/httpd_connection_handler.e]: this is in ancestor, but right I will add a -- <Precursor>

File: library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e (hunk 10 above)
Comment (jocelyn) [inline library/server/ewsgi/connectors/standalone/src/httpd/concurrency/scoop/pool/concurrent_pool.e]: should be set_capacity, but set_count is an acceptable name for now.

**Comment by jocelyn** [comment]
Any comment on the concurrent design itself? and the breaking change coming with this move?
(see google group messages for other discussions)

**Comment by jocelyn** [comment]
This will be merged as v1, and become the version of the master

**Pull request closed without merging**